ARCHITECTURE.md

```diff
@@ -0,0 +1,97 @@
+
+The `Dynatrace Operator` is not a single Pod, it consists of multiple components, encompassing several Kubernetes concepts.
+
+#### Operator
+This component/pod is the one that _reacts to_ the creation/update/delete of our`CustomResource(s)`, causing the `Operator` to _reconcile_.
+A _reconcile_ just means that it will check what is in the `CustomResource(s)` and according to that creates/updates/deletes resources in the Kubernetes environment. (So the state of the Kubernetes Environment matches the state described in the `CR`)
+
+Relevant links:
+- [Operator Pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+#### Webhook
+This component/pod is the one that _intercepts_ creation/update/delete of Kubernetes Resources (only those that are relevant), then either mutates or validates them.
+- Validation: We only use it for our `CustomResource(s)`, it's meant to catch known misconfigurations. If the validation webhook detects a problem, the user is warned, the change is denied and rolled back, like nothing happened.
+- Mutation: Used to modify Kubernetes Resources "in flight", so the Resource will be created/updated in the cluster like if it was applied with added modifications.
+   - We have 2 use-cases for this:
+      - Seamlessly modifying user resources with the necessary configuration needed for Dynatrace observability features to work.
+      - Handle time/timing sensitive minor modifications (labeling, annotating) of user resources, which is meant to help the `Operator` perform more reliably and timely.
+
+Relevant links:
+- [What are webhooks?](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks)
+
+#### Init Container
+Some configurations need to happen on the container filesystem level, like setting up a volume or creating/updating configuration files.
+To achieve this we add our init-container (using the `Webhook`) to user Pods. As init-containers run before any other container, we can setup the environment of user containers to enable Dynatrace observability features.
+
+Relevant links:
+- [Init Containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+#### CSI-Driver
+A component that is present on all nodes, meant to provide volumes (based on the node's filesystem) to make the capabilities provided by the `Operator` to use less disk space and be more performant.
+
+Relevant links:
+- [CSI volume](https://kubernetes.io/docs/concepts/storage/volumes/#csi)
+
+## Code Map
+> TODO: Improve folder structure before documenting it more deeply, as its kind of a mess now. If I didn't mention it now, then I probably don't like its current location.
+
+### `config`
+Contains the `.yaml` files that are need to deploy the `Operator` and it`s components into a Kubernetes cluster.
+- most `.yaml` files are part of the Helm chart
+- other `.yaml` files are relevant for different marketplaces
+### `hack`
```

Comment:
hack make is missing for me, as it contains a lot of information and commands that are needed to develop